### PR TITLE
fix(react-storage): default nullish progress to 100% if status is "complete"

### PR DESCRIPTION
**Pull request by jordanvn**
#### Description of changes
* In storage browser actions where progress is not defined yet there is a `COMPLETE` status (such as uploading a zero-byte file), default to setting the progress to 100% instead of 0%

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/getActionViewTableData.ts [hunk 1]
```diff
@@ -169,7 +169,9 @@ export const getActionViewTableData = <T extends TaskData = TaskData>({
               type: 'number',
               content: {
                 value: progress,
-                displayValue: `${getPercentValue(progress ?? 0)}%`,
+                displayValue: `${getPercentValue(
+                  progress ?? status === 'COMPLETE' ? 1 : 0
```
[the remark this answers is not in the document]
Comment (jordanvn) [reply packages/react-storage/src/components/StorageBrowser/views/LocationActionView/getActionViewTableData.ts:174]: Updated to add comment "Zero-byte files do not allow progress to be tracked, only status". Do you think this sufficiently explains the circumstances for the change?

[the remark this answers is not in the document]
Comment (jordanvn) [reply packages/react-storage/src/components/StorageBrowser/views/LocationActionView/getActionViewTableData.ts:175]: Great point, I've updated the comment to match your suggestion

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/getActionViewTableData.ts [hunk 2]
```diff
@@ -169,7 +169,11 @@ export const getActionViewTableData = <T extends TaskData = TaskData>({
               type: 'number',
               content: {
                 value: progress,
-                displayValue: `${getPercentValue(progress ?? 0)}%`,
+                displayValue: `${getPercentValue(
+                  // Zero-byte files are uploaded instantly and don't report a progress
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationActionView/getActionViewTableData.ts:173]: Would drop this line from the comment:
- a custom upload action may not exhibit this behavior
- `progress` may not always be upload specific

**Review by calebpollman — COMMENTED** [review]
Minor feedback on a comment, other than that lgtm

**Review by AllanZhengYP — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/getActionViewTableData.ts (hunk 2 above)
Comment (jordanvn) [reply packages/react-storage/src/components/StorageBrowser/views/LocationActionView/getActionViewTableData.ts:173]: Sounds reasonable to me. Will update to just include note that we're defaulting to 100% progress if value is unavailable but status is complete

**Review by AllanZhengYP — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**